### ISSUE TITLE
URL shortening does not work anymore ? !

### ISSUE THREAD

**Issue by sk4zuzu** [lidel/yt-looper#66]
Just noticed this problem. :/

**Comment by lidel** [comment]
Indeed, does not work at http://yt.aergia.eu and it is at https://github.com/lidel/yt-looper/commit/a9ff93fe7271a2838c91f52254fc99840f06a60f

**Comment by lidel** [comment]
I did not investigate it yet, but if we got blocked by Google, we should switch to http://42.pl/u/.
This shortener has been alive for over a decade and provides super-simple API for automated shortening:

`GET http://42.pl/u/?auto=1&url=yoururl`

**Comment by sk4zuzu** [comment]
:+1:

**Comment by lidel** [comment]
Yup, Google blocked us due to high daily usage :-)

> Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup.

```javascript
{"readyState":4,"responseText":"{\n \"error\": {\n  \"errors\": [\n   {\n    \"domain\": \"usageLimits\",\n    \"reason\": \"dailyLimitExceededUnreg\",\n    \"message\": \"Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup.\",\n    \"extendedHelp\": \"https://code.google.com/apis/console\"\n   }\n  ],\n  \"code\": 403,\n  \"message\": \"Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup.\"\n }\n}\n","responseJSON":{"error":{"errors":[{"domain":"usageLimits","reason":"dailyLimitExceededUnreg","message":"Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup.","extendedHelp":"https://code.google.com/apis/console"}],"code":403,"message":"Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup."}},"status":403,"statusText":"Forbidden"}"
```

I don't like the idea of registering and managing API keys outside of repo, so I'll switch to other provider for now.

**Comment by lidel** [comment]
Turns out 42.pl does not send CORS header, so I can't use it from JS, it works only from curl:
```bash
> curl "http://u.42.pl/?auto=1&url=http://google.com"
http://u.42.pl/1B
```
Maybe we should stay with Google. I see that it may be a bug on their side, as people are having the same issue:
https://stackoverflow.com/questions/29267628/url-shortener-api-limit-exceeded-code-403

**Comment by sk4zuzu** [comment]
Very nice indeed. :hammer:

**Comment by lidel** [comment]
Ok, so it was possible to get a key that works only with URL shortening service and only from a specific domain. That is why it will work at http://yt.aergia.eu but not on localhost.

If one wants to run looper under different host, a new key can be generated at https://console.developers.google.com

**tl;dr:** with this key we have limits that should be enough:

![2015-03-26-202737_337x302_scrot](https://cloud.githubusercontent.com/assets/157609/6855506/2ea0ad20-d3f9-11e4-88e3-c8356c088371.png)
